pkg/common/utils/ioutil_test.go

```diff
@@ -18,12 +18,86 @@ package utils
 
 import (
 	"bytes"
+	"io"
 	"testing"
 
 	"github.com/cloudwego/hertz/pkg/common/test/assert"
 	"github.com/cloudwego/hertz/pkg/network"
 )
 
+type writeReadTest interface {
+	Write(p []byte) (n int, err error)
+	Malloc(n int) (buf []byte, err error)
+	WriteBinary(b []byte) (n int, err error)
+	Flush() error
+}
+
+type readerTest interface {
+	ReadFrom(r io.Reader) (n int64, err error)
+	Malloc(n int) (buf []byte, err error)
+	WriteBinary(b []byte) (n int, err error)
+	Flush() error
+}
+
+type testWriter struct {
+	w io.Writer
+}
+
+func (t testWriter) Write(p []byte) (n int, err error) {
+	return
+}
+
+func (t testWriter) Malloc(n int) (buf []byte, err error) {
+	return
+}
+
+func (t testWriter) WriteBinary(b []byte) (n int, err error) {
+	return
+}
+
+func (t testWriter) Flush() error {
+	return nil
+}
+
+type testReader struct {
+	r io.ReaderFrom
+}
+
+func (t testReader) ReadFrom(r io.Reader) (n int64, err error) {
+	return
+}
+
+func (t testReader) Malloc(n int) (buf []byte, err error) {
+	return
+}
+
+func (t testReader) WriteBinary(b []byte) (n int, err error) {
+	return
+}
+
+func (t testReader) Flush() error {
+	return nil
+}
+
+func newTestWriter(w io.Writer) writeReadTest {
+	return &testWriter{
+		w: w,
+	}
+}
+
+func newTestReaderForm(r io.ReaderFrom) readerTest {
+	return &testReader{
+		r: r,
+	}
+}
+
+func newLimitReader(r *bytes.Buffer) io.LimitedReader {
```

Comment:
Can it be reused or move it to mock dir so that other developers can use it in the future

Reply:
of course, i have move it to mock dir